src/main/java/uk/gov/hmcts/reform/userprofileapi/infrastructure/clients/IdamUserResponse.java

```diff
@@ -0,0 +1,91 @@
+package uk.gov.hmcts.reform.userprofileapi.infrastructure.clients;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import javax.validation.constraints.NotNull;
+
+public class IdamUserResponse {
+    private Boolean active;
+    @NotNull
+    private String email;
+    private String forename;
+    private String id;
+
+    @JsonCreator
+    public IdamUserResponse(@JsonProperty(value = "active") Boolean active,
+                               @NotNull @JsonProperty(value = "email") String email,
+                               @JsonProperty(value = "forename") String forename,
+                               @JsonProperty(value = "id") String id,
+                               @JsonProperty(value = "locked") Boolean locked,
+                               @JsonProperty(value = "roles") List<String> roles,
+                               @JsonProperty(value = "surname") String surname) {
+        this.active = active;
+        this.email = email;
+        this.forename = forename;
+        this.id = id;
+        this.locked = locked;
+        this.roles = roles;
+        this.surname = surname;
+    }
+
+    private Boolean locked;
+    private List<String> roles;
+    private String surname;
+
+    public Boolean getActive() {
+        return active;
+    }
+
+    public void setActive(Boolean active) {
+        this.active = active;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public String getForename() {
+        return forename;
+    }
+
+    public void setForename(String forename) {
+        this.forename = forename;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {

```

Comment:
Why are we using setters if the values have already been assigned in the constructor?

Reply:
Removed